testConnectivity.py

```diff
@@ -409,75 +436,157 @@ def testConnectivity(args):
-                # Initial phase2Write will be phase-1 if 3 consecutive phases are good nScan times
-                # This will be overwritten to phase-2 if a 4th consecutive phase is found to be good
-                # This will again be overwritten to phase-2 if a 5th consecutive phase is found to be good
-                # After 5 consecutive phases are good the procedure will exit
-                for phase in range(0,16):
-                    phaseRes_idxm0 = dict_phaseScanResults[ohN][vfat*16+phase]
-
-                    if (    phaseRes_idxm4 == args.nPhaseScans and
-                            phaseRes_idxm3 == args.nPhaseScans and
-                            phaseRes_idxm2 == args.nPhaseScans and 
-                            phaseRes_idxm1 == args.nPhaseScans and 
-                            phaseRes_idxm0 == args.nPhaseScans): # Found a sweet spot
-                        phase2Write = phase-2
-                        break
-                    elif (  phaseRes_idxm3 == args.nPhaseScans and
-                            phaseRes_idxm2 == args.nPhaseScans and
-                            phaseRes_idxm1 == args.nPhaseScans and
-                            phaseRes_idxm0 == args.nPhaseScans): 
-                        phaseRes_idxm4 = phaseRes_idxm3
-                        phaseRes_idxm3 = phaseRes_idxm2
-                        phaseRes_idxm2 = phaseRes_idxm1
-                        phaseRes_idxm1 = phaseRes_idxm0
-                        phase2Write = phase-2
-                    elif (  phaseRes_idxm2 == args.nPhaseScans and
-                            phaseRes_idxm1 == args.nPhaseScans and
-                            phaseRes_idxm0 == args.nPhaseScans): 
-                        phaseRes_idxm3 = phaseRes_idxm2
-                        phaseRes_idxm2 = phaseRes_idxm1
-                        phaseRes_idxm1 = phaseRes_idxm0
-                        phase2Write = phase-1
-                    elif (phaseRes_idxm1 == args.nPhaseScans and phaseRes_idxm0 == args.nPhaseScans): # Last phase and this phase are good
-                        phaseRes_idxm2 = phaseRes_idxm1
-                        phaseRes_idxm1 = phaseRes_idxm0
-                    elif (phaseRes_idxm0 == args.nPhaseScans): # Only this phase is good
-                        phaseRes_idxm1 = phaseRes_idxm0
-                    else: # Reset
-                        phaseRes_idxm0 = 0
-                        phaseRes_idxm1 = 0
-                        phaseRes_idxm2 = 0
-                        phaseRes_idxm3 = 0
-                        phaseRes_idxm4 = 0
+                phaseCounts = np.array([ dict_phaseScanResults[ohN][vfat*GBT_PHASE_RANGE+ph] for ph in range(0,GBT_PHASE_RANGE) ])
+                allBadPhases = np.where(phaseCounts!=args.nPhaseScans)[0]
+                badPhaseCounts = np.delete(allBadPhases,np.where(allBadPhases==15)[0]) ## remove 15 from the list of bad phases
+                phaseSum = 0
+                if len(badPhaseCounts) == 0:
```

Comment:
maybe I misunderstood but I thought this block was being removed in favor of the window method around a dead phase?

Reply:
I don't see the block that simplifies this...maybe I missed it though; or is there some more git foo confusion going on?

Reply:
Nevermind I see the `True pass statement`